### PR TITLE
Merge GenerateGlobalAnalyzerConfigs.csproj into GenerateDocumentation…

### PR DESCRIPTION
**Pull request by mavasani**
…AndConfigFiles.csproj

Now we have single post-build utility tool 'GenerateDocumentationAndConfigFiles' for generating all of the following files:
1. Documentation files (md and sarif files + file listing rules missing documentation)
2. Config files (global configs for AnalysisLevel and AnalysisMode + editorconfig/ruleset files)
3. props and targets files shipping with the packages

### CODE REVIEW

**Comment by mavasani** [comment]
Also tagging @Youssef1313 who has helped improving the post-build tooling.

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 1]
```diff
@@ -589,6 +609,159 @@ static bool checkHelpLink(string helpLink)
                     }
                 }
             }
+
```
Comment (mavasani) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:612]: All the added code here moved as-is from the deleted Program.cs in GenerateGlobalAnalyzerConfigs.csproj.

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 2]
```diff
@@ -890,6 +1063,363 @@ private static void Validate(string fileWithPath, string fileContents, List<stri
...
+            string editorconfigFileName,
+            string editorconfigTitle,
+            string editorconfigDescription,
+            AnalysisMode analysisMode,
+            SortedList<string, DiagnosticDescriptor> sortedRulesById,
+            (ImmutableArray<ReleaseTrackingData> shippedFiles, Version version) shippedReleaseData)
+        {
+            Debug.Assert(editorconfigFileName.EndsWith(".editorconfig", StringComparison.Ordinal));
+
+            var text = GetEditorconfigText(
```
Comment (mavasani) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:1077]:
```suggestion
            var text = GetGlobalconfigText(
```

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 3]
```diff
@@ -890,6 +1063,363 @@ private static void Validate(string fileWithPath, string fileContents, List<stri
...
+                analysisMode,
+                sortedRulesById,
+                shippedReleaseData);
+            var directory = Directory.CreateDirectory(folder);
+            var editorconfigFilePath = Path.Combine(directory.FullName, editorconfigFileName);
+            File.WriteAllText(editorconfigFilePath, text);
+            return;
+
+            // Local functions
+            static string GetEditorconfigText(
```
Comment (mavasani) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:1089]:
```suggestion
            static string GetGlobalconfigText(
```

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 4]
```diff
@@ -890,6 +1063,363 @@ private static void Validate(string fileWithPath, string fileContents, List<stri
...
+            // Local functions
+            static string GetGlobalconfigText(
+                string editorconfigTitle,
+                string editorconfigDescription,
+                AnalysisMode analysisMode,
+                SortedList<string, DiagnosticDescriptor> sortedRulesById,
+                (ImmutableArray<ReleaseTrackingData> shippedFiles, Version version)? shippedReleaseData)
+            {
+                var result = new StringBuilder();
+                StartEditorconfig();
```
Comment (mavasani) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:1097]:
```suggestion
                StartGlobalconfig();
```

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 5]
```diff
@@ -890,6 +1063,363 @@ private static void Validate(string fileWithPath, string fileContents, List<stri
...
+                AddRules(analysisMode);
+                return result.ToString();
+
+                void StartEditorconfig()
```
Comment (mavasani) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:1101]:
```suggestion
                void StartGlobalconfig()
```

**Review by Evangelink — APPROVED** [review]
LGTM

**Review by Youssef1313 — COMMENTED** [review]
Later on, we may consider splitting the single large class into a smaller, more manageable & maintainable classes. That's not very important if the logic here isn't expected to change a lot.

**Comment by mavasani** [comment]
> Later on, we may consider splitting the single large class into a smaller, more manageable & maintainable classes

Yes, that is something we can consider in future based on how this works out.

**Comment by mavasani** [comment]
Thanks @Evangelink!

**Pull request merged**